### PR TITLE
mdoc 0.3.1 (new formula)

### PR DESCRIPTION
**Pull request by kmaasrud**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

MDoc is a document environment and command line tool for producing academic documents. It aims at being a simpler version of a LaTeX distributions that allows you to use Markdown and handles a lot of the nitty-gritty for you.

### CODE REVIEW

File: Formula/mdoc.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  depends_on "libpng"
+  depends_on "openssl@1.1"
+  depends_on "pandoc"
+  uses_from_macos "icu4c"
+
+  def install
+    # From https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/tectonic.rb
+    ENV.cxx11
+    ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version # needed for CLT-only builds
+    ENV.delete("HOMEBREW_SDKROOT") if MacOS.version == :high_sierra
```
Comment (SMillerDev) [inline Formula/mdoc.rb:28]: Why?

File: Formula/mdoc.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  homepage "https://kmaasrud.com/mdoc"
+  url "https://github.com/kmaasrud/mdoc/archive/0.3.1.tar.gz"
+  sha256 "6807701201e0149bdb09d2ca7f32590a646dfb238823a5f448384a844e8b78f1"
+  license "MIT"
+  head "https://github.com/kmaasrud/mdoc.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
```
Comment (SMillerDev) [inline Formula/mdoc.rb:12]: There don't seem to be any pre-releases so tag checking seems good enough.

File: Formula/mdoc.rb (hunk 1 above)
Comment (kmaasrud) [reply Formula/mdoc.rb:28]: MDoc depends on Tectonic, which needs quite a few external C libraries included. This is to ensure they are properly linked to when building. As mentioned in the comment, I found this method from the Tectonic formula, and it worked fine on my computer 😄

File: Formula/mdoc.rb (hunk 2 above)
Comment (kmaasrud) [reply Formula/mdoc.rb:12]: Do you want me to use the regex method like this?

```ruby
livecheck do
  url :stable
  regex(/^v?(\d+(?:\.\d+)+)\.t/i)
end
```

Comment (SMillerDev) [reply Formula/mdoc.rb:12]: Don't specify anything and use the default

File: Formula/mdoc.rb (hunk 1 above)
Comment (carlocab) [reply Formula/mdoc.rb:23]: Which external C libraries are these?

Comment (kmaasrud) [reply Formula/mdoc.rb:23]: The dependencies are documented [here](https://tectonic-typesetting.github.io/book/latest/howto/build-tectonic/index.html#third-party-dependencies). You can see the Tectonic guys doing the same on [their formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tectonic.rb).

Comment (carlocab) [reply Formula/mdoc.rb:23]: What happens when you don't do this? I can see the `tectonic` formula does the same, but I'm not even sure `tectonic` should be doing that at all.

**Comment by carlocab** [comment]
This looks like a nice piece of software, but it currently does not meet the notability requirements for this repository. The CI log reads
```
  mdoc:
    * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
```
You will also see this error when you run `brew audit --new mdoc`.

If you wish, you can host this formula in your own tap. Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

We can consider this formula again when it meets our notability requirements. If you do choose to contribute it in the future, please address any pending feedback from above.

**Comment by kmaasrud** [comment]
Oh! Oh well, I'll try again when I get some clout 😎

**Pull request closed without merging**